### PR TITLE
test: repro for #2662: extend import list miss separator

### PR DESCRIPTION
**Pull request by guibou**
This is a repro for #2662.

I've tested with GHC 8.10.7, 9.0.2 and 9.2.1, using nix:

(Note: I do have to `export PATH` so it point to the correct `ghcide`. Is that the intended workflow or I missed something?)

GHC 9.2.1:

The import part is:

```diff
- "module ModuleB where\nimport Data.List.NonEmpty (fromList, NonEmpty ((:|)))\nmain = case (fromList []) of _ :| _ -> pure ()\n"
+ "module ModuleB where\nimport Data.List.NonEmpty (fromListNonEmpty ((:|)))\nmain = case (fromList []) of _ :| _ -> pure ()\n"
```

```
$ nix-shell -p haskell.compiler.ghc921 -p cabal-install -p zlib    

[nix-shell:~/srcs/haskell-language-server]$ export PATH=/home/guillaume/srcs/haskell-language-server/dist-newstyle/build/x86_64-linux/ghc-9.2.1/ghcide-1.6.0.0/x/ghcide/build/ghcide/:$PATH

[nix-shell:~/srcs/haskell-language-server]$ ls /home/guillaume/srcs/haskell-language-server/dist-newstyle/build/x86_64-linux/ghc-9.2.1/ghcide-1.6.0.0/x/ghcide/build/ghcide/
autogen  ghcide  ghcide-tmp

[nix-shell:~/srcs/haskell-language-server]$ cabal run --project-file ./cabal-ghc921.project ghcide:ghcide-tests -- -p 'infix constru'
Up to date
ghcide
  code actions
    extend import actions
      with checkAll
        extend single line import with infix constructor: FAIL (1.10s)
          test/exe/Main.hs:1827:
          expected: "module ModuleB where\nimport Data.List.NonEmpty (fromList, NonEmpty ((:|)))\nmain = case (fromList []) of _ :| _ -> pure ()\n"
           but got: "module ModuleB where\nimport Data.List.NonEmpty (fromListNonEmpty ((:|)))\nmain = case (fromList []) of _ :| _ -> pure ()\n"
          Use -p '/infix constru/&&/with checkAll.extend single line import with infix constructor/' to rerun this test only.
      without checkAll
        extend single line import with infix constructor: FAIL (1.11s)
          test/exe/Main.hs:1827:
          expected: "module ModuleB where\nimport Data.List.NonEmpty (fromList, NonEmpty ((:|)))\nmain = case (fromList []) of _ :| _ -> pure ()\n"
           but got: "module ModuleB where\nimport Data.List.NonEmpty (fromListNonEmpty ((:|)))\nmain = case (fromList []) of _ :| _ -> pure ()\n"
          Use -p '/infix constru/&&/without checkAll.extend single line import with infix constructor/' to rerun this test only.

2 out of 2 tests failed (2.22s)
```

GHC 9.0:

```
$ nix-shell -p haskell.compiler.ghc902 -p cabal-install -p zlib

[nix-shell:~/srcs/haskell-language-server]$ export PATH=/home/guillaume/srcs/haskell-language-server/dist-newstyle/build/x86_64-linux/ghc-9.0.2/ghcide-1.6.0.0/x/ghcide/build/ghcide:$PATH

[nix-shell:~/srcs/haskell-language-server]$ which ghcide
/home/guillaume/srcs/haskell-language-server/dist-newstyle/build/x86_64-linux/ghc-9.0.2/ghcide-1.6.0.0/x/ghcide/build/ghcide/ghcide

[nix-shell:~/srcs/haskell-language-server]$ cabal run --project-file ./cabal-ghc90.project ghcide:ghcide-tests -- -p 'infix constru'
Up to date
ghcide
  code actions
    extend import actions
      with checkAll
        extend single line import with infix constructor: OK (1.16s)
      without checkAll
        extend single line import with infix constructor: OK (1.14s)

All 2 tests passed (2.31s)
```

GHC 8.10.7:

```
λ narwal haskell-language-server → λ git repro_2662 → nix-shell -p haskell.compiler.ghc8107 -p cabal-install -p zlib

[nix-shell:~/srcs/haskell-language-server]$ export PATH=/home/guillaume/srcs/haskell-language-server/dist-newstyle/build/x86_64-linux/ghc-8.10.7/ghcide-1.6.0.0/x/ghcide/build/ghcide:$PATH
[nix-shell:~/srcs/haskell-language-server]$ which ghcide
/home/guillaume/srcs/haskell-language-server/dist-newstyle/build/x86_64-linux/ghc-8.10.7/ghcide-1.6.0.0/x/ghcide/build/ghcide/ghcide

[nix-shell:~/srcs/haskell-language-server]$ cabal run --project-file ghcide:ghcide-tests -- -p 'infix constru'
The given project file 'ghcide:ghcide-tests' does not exist.

[nix-shell:~/srcs/haskell-language-server]$ cabal run ghcide:ghcide-tests -- -p 'infix constru'
Resolving dependencies...
Up to date
ghcide
  code actions
    extend import actions
      with checkAll
        extend single line import with infix constructor: OK (1.06s)
      without checkAll
        extend single line import with infix constructor: OK (1.05s)

All 2 tests passed (2.11s)
```

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2664"></a>

### CODE REVIEW

File: ghcide/test/exe/Main.hs [hunk 1]
```diff
@@ -1520,6 +1520,36 @@ extendImportTests = testGroup "extend import actions"
                     , "import ModuleA as A (stuffB, (.*))"
                     , "main = print (stuffB .* stuffB)"
                     ])
+        , testSession "extend single line import with infix constructor" $ template
```
Comment (jneira) [inline ghcide/test/exe/Main.hs:1523]: a nice way to merge it before the fix is annotate the tests with a knownBroken for a specific ghc version, like other tests in the file

**Review by jneira — COMMENTED** [review]
thanks for the regression tests, they will make the fix lot easier

File: ghcide/test/exe/Main.hs (hunk 1 above)
Comment (jneira) [reply ghcide/test/exe/Main.hs:1523]: for example 
https://github.com/haskell/haskell-language-server/blob/016ff739c4be0dcb50e7a8575a4e8fefe3a58d5a/ghcide/test/exe/Main.hs#L2639

Comment (guibou) [reply ghcide/test/exe/Main.hs]: Done, thank you.

**Review by jneira — APPROVED** [review]
perfect, many thanks, maybe is asking for too much but have you any plans to tackle the error itself? 🙂

**Comment by guibou** [comment]
> perfect, many thanks, maybe is asking for too much but have you any plans to tackle the error itself? slightly_smiling_face

Well, not yet. Actually, I discovered the issue actually working on another issue (fixing the eval plugin). I'll have a look at it tomorrow ;)

**Pull request merged**